server/plugin.go

```diff
@@ -280,6 +280,11 @@ func (p *Plugin) stop() {
 	if p.monitor != nil {
 		p.monitor.Stop()
 	}
+	if p.metricsServer != nil {
+		if err := p.metricsServer.Shutdown(); err != nil {
+			p.API.LogWarn("Error shutting down metrics server", "error", err)
```

Comment:
Funny how we log an error as a warning. I sort of understand the reasoning, we are shutting down after all, but find it interesting nonetheless.

Reply:
My own rationale is ERROR = "a human must take some action". Despite an error having occurred in this case, given we're shutting down anyway, no human need get involved. BUT where it gets interesting is also that a "spike" in WARN logs is also something a human should investigate. Next up for MS Teams is a holistic review of all logging to align to this philosophy, and then enable Grafana alerts for all ERROR logs and spikes in WARN logs. (Ambitious, I know! We'll see where we land :p)